reactive-grpc-common/src/main/java/com/salesforce/reactivegrpccommon/ReactiveConsumerStreamObserver.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright (c) 2017, salesforce.com, inc.
+ *  All rights reserved.
+ *  Licensed under the BSD 3-Clause license.
+ *  For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package com.salesforce.reactivegrpccommon;
+
+import com.google.common.base.Preconditions;
+import io.grpc.Status;
+import io.grpc.stub.ClientCallStreamObserver;
+import io.grpc.stub.ClientResponseObserver;
+import org.reactivestreams.Publisher;
+
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * ReactorConsumerStreamObserver configures client-side manual flow control for the consuming end of a message stream.
+ *
+ * @param <TRequest>
+ * @param <TResponse>
+ */
+public abstract class ReactiveConsumerStreamObserver<TRequest, TResponse> implements ClientResponseObserver<TRequest, TResponse> {
+    private ReactiveStreamObserverPublisher<TResponse> publisher;
+    private Publisher<TResponse> rxConsumer;
+    private CountDownLatch beforeStartCalled = new CountDownLatch(1);
+
+    public abstract Publisher<TResponse> getReactiveConsumerFromPublisher(ReactiveStreamObserverPublisher<TResponse> publisher);
```

Comment:
I'm not sure about the name... Other ideas ?

Reply:
🤷‍♂️ Works for me.